packages/leaflet-map/src/base-map.tsx

```diff
@@ -0,0 +1,262 @@
+  scrollWheelZoom = true,
+	center = {
+		lat: 52.37104644463586,
+		lng: 4.900402911007405,
+	},
+
+  tilesVariant = 'nlmaps',
+  tiles = undefined,
+  minZoom = 0,
+  maxZoom = 25,
+
+  categorize = undefined,
+  
+  // ToDo: search = false,
+	zoomposition = 'bottomleft',
+	disableDefaultUI = true,
+  clustering = {
+    isActive: true,
+  },
+
+  onClick = undefined,
+  onMarkerClick = undefined,
+
+  ...props
+}: PropsWithChildren<BaseMapWidgetProps>) {
+
+  let [, setMapIsReady] = useState(false);
+  let [currentMarkers, setCurrentMarkers] = useState(markers);
+
+  let [mapId] = useState(`${parseInt(Math.random() * 1e8 as any as string)}`);
+  let [mapRef] = useMapRef(mapId);
+
+  // map is ready
+  useEffect(() => {
+		setMapIsReady(true);
+		let event = new CustomEvent('osc-map-is-ready', { detail: { id: mapId } });
+		window.dispatchEvent(event);
+  }, []);
+
+	// auto zoom and center on init
+  useEffect(() => {
+    if (!mapRef) return;
+	  if (autoZoomAndCenter) {
+		  let centerOn = ( autoZoomAndCenter == 'area' && area ) || ( currentMarkers && currentMarkers.length && currentMarkers );
+      if (!centerOn) {
+				centerOn = area;
+      }
+		  if (centerOn) {
+			  setBoundsAndCenter( centerOn );
+		  }
+	  }
+  }, [mapRef]);
+
+  // update center
+  useEffect(() => {
+    if (!mapRef) return;
+		if (center) {
+			setBoundsAndCenter(center);
+		}
+  }, [center]);
+
+  // markers
+  useEffect(() => {
+
+    if (markers.length == 0 && currentMarkers.length == 0) return;
+
+    let result = [...markers];
+
+    result.map((marker, i) => {
+
+      // add
+      let markerData: MarkerProps = {...marker};
+      markerData.markerId = `${parseInt(Math.random() * 1e8 as any as string)}`;
+
+      // iconCreateFunction
+      markerData.iconCreateFunction = markerData.iconCreateFunction || iconCreateFunction;
+
+      // categorize
+      if (categorize?.categorizeByField && markerData.data?.[categorize.categorizeByField]) {
+        let type = markerData.data?.[categorize.categorizeByField];
+        let category = categorize.categories?.[type];
+        if (category) {
+          if (!markerData.icon) {
+            let icon = category.icon;
+            if (!icon && category.color) {
+              icon = {...defaultIcon, color: category.color};
+            }
+            if (icon) {
+              markerData.icon = icon;
+            }
+          }
+        }
+      }
+
+      // fallback on defaultIcon
+      if (!markerData.icon && defaultIcon) {
+        markerData.icon = defaultIcon;
+      }
+
+      markerData.onClick = markerData.onClick ? [ ...markerData.onClick, onMarkerClick ] : [ onMarkerClick ];
+
+      // ToDo
+      markerData.isVisible = true;
+
+		  if (clustering && clustering.isActive && !markerData.doNotCluster) {
+        markerData.isClustered = true;
+		  }
+
+      result[i] = markerData
+
+    })
+
+    setCurrentMarkers(result);
+
+  }, [markers]);
+
+  async function setBoundsAndCenter(points: Object) {
```

Comment:
Het type van points staat op Object, maar dit wordt in de functie alleen gebruikt als een array. Kan dit niet naar een Array<T> type worden gezet?

Reply:
Goed punt. Ik heb de call in useEffect(center) daar ook op aangepast.

Reply:
Nu met een heel eigen type